Documentation/Architecture.md

```diff
@@ -0,0 +1,102 @@
+
+Another essential task is interaction with user input and handling new lines, deleting list items, and indents around:
+ - Lists
+ - Quote
+ - Paragraphs
+
+The following "ensure..." methods handle all these scenarios:
+ - ['ensureRemovalOfParagraphAttributesWhenPressingEnterInAnEmptyParagraph'](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/TextView.swift#L1945)
+ - [ensureInsertionOfEndOfLine](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/TextView.swift#L697)
+ - [evaluateRemovalOfSingleLineParagraphAttributesAfterSelectionChange](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/TextView.swift#L1884)
+
+These methods check if you and adding or removing new paragraphs around lists or quotes and update the attributes accordingly. 
+
+# Presentation of the custom attributes
+
+The [LayoutManager](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/LayoutManager.swift) class handles the presentation for the custom attributes that the standard [NSLayoutManager](https://developer.apple.com/documentation/uikit/nslayoutmanager) is unable to do:
+
+ - Backgrounds and vertical bars for Quotes: `drawBlockquotes`
+ - Backgrounds for Pre: `drawHTMLPre`
+ - List bullets: `drawLists`
+
+The layout manager uses the extra information provided by the [ParagraphStyle](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/TextKit/ParagraphStyle.swift) class to be able to render the correct background and bullets for each element.
+
+# HTML Converter
+
+The [HTMLConverter](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/NSAttributedString/Conversions/HTMLConverter.swift) class handles all the conversation between HTML and NSAttributedString. The main entry points are: 
+ - [attributedString(from:html, defaultAttributes)](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/NSAttributedString/Conversions/HTMLConverter.swift#L58), converts from HTML to a NSAttributedString object
+ - [html(from:NSAttributedString, pretify)](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/NSAttributedString/Conversions/HTMLConverter.swift#L110) converts from NSAttributedString to HTML
+
+# Converting from HTML to NSAttributedString
+
+Two classes are responsible for the conversion:
+ - [HTMLParser](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/Libxml2/Converters/In/HTMLParser.swift) transforms from HTML text to an in-memory XML tree.
+ - [AttributedStringSerializer](https://github.com/wordpress-mobile/AztecEditor-iOS/blob/develop/Aztec/Classes/NSAttributedString/Conversions/AttributedStringSerializer.swift) converts from the in-memory tree to NSAttributedString.
+
+<img src="resources/html_to_nsattributedstring.png">
```

Comment:
The images show like a `?` square to me. Not sure if this will work after the merge, or if we need to add the full url for the image.

I see the images being added on this PR, so maybe it will fix itself after merging? :)

Reply:
They will show up :)